### ISSUE TITLE
The inserter search doesn't work when dictating with speech recognition software

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#6010]
Breaking this down from Eric Wright (@ewaccess) testing session with Dragon NaturallySpeaking (speech recognition software). Please refer to the videorecorded session posted here:
https://make.wordpress.org/accessibility/2018/03/28/accessibility-of-gutenberg-the-state-of-play/

Please refer to the video above, starting from about min. 13:40

When dictating text in the inserter search field, the search isn't triggered. Not sure why, but this could be perhaps related to the event used on the search field. See also https://github.com/WordPress/gutenberg/issues/5468 which I guess is a similar issue, where it seems the UI doesn't get the text in the post title field has changed.

**Comment by afercia** [comment]
See also https://github.com/WordPress/gutenberg/issues/5468 when this issue was first noticed on the Title field.

**Comment by afercia** [comment]
Removing from the Merge Proposal milestone, as right now it's unclear how to solve this and it should be solved together with the similar issues anyways.

**Comment by enriquesanchez** [comment]
I just ran a quick test using VoiceControl on Safari and I was able to dictate into the search field. I was also able to add an edit the post tile (#5468).

Is this issue still happening with Dragon?

**Comment by afercia** [comment]
Needs to be tested specifically with Dragon on Windows. Maybe trying to involve some Dragon users could be an item to discuss during next accessibility weekly meeting.

**Comment by enriquesanchez** [comment]
I tested with Dragon Home 15.3 on the Gutenberg plugin and I'm able to reproduce this issue. I can dictate into the search field but my dictation does not fire the "live search".

**Comment by mapk** [comment]
@tellthemachines is this something you can look into?

**Comment by t-hamano** [comment]
It's been a while since this issue was reported, but can you still reproduce this issue?

I don't have the Dragon app, but I tested it using [the voice typing feature in Windows OS](https://support.microsoft.com/en-us/windows/use-voice-typing-to-talk-instead-of-type-on-your-pc-fec94565-c4bd-329d-e59a-af033fa5689f). There doesn't seem to be any problem now.

Please note that in the video below, I changed the language of WordPress to Japanese because my voice was detected as Japanese:

https://github.com/user-attachments/assets/c4029295-708d-4d0d-9572-ee6559f7595a

Could someone test whether this issue can be reproduced with the Dragon app?

**Comment by afercia** [comment]
@t-hamano thanks for looking into this after such a long time.
I can't test with Dragon (a license is pretty expensive). Just noting that the original reported issue is very specific to Dragon because of the special way Dragon used to modify the input `value`. For more details, see https://github.com/WordPress/gutenberg/issues/5468#issuecomment-391870958

I will ping the WordPress accessibility team to see if there's anyone available to test with Dragon.

**Comment by joedolson** [comment]
I just tested this with Dragon Professional 16, and it worked as expected; the search query executed, and the search results appeared and could be navigated to. So I think that this is working now, at least with the latest version of Dragon. 

I tested in the current 6.7.1 without Gutenberg and in the 6.8 alpha with Gutenberg 19.9.0 active.

**Comment by afercia** [comment]
Thanks @joedolson ! 
For history: Dragon used to directly set the `.value` of the input field and didn't trigger keyboard events. 

I was able to find some old Dragon documentation where it's mentioned that Dragon used some Windows API to change the text by sending Windows messages via `WM_SETTEXT` and `EM_REPLACESEL`. In such a scenario, no keyboard events are triggered. It's worth reminding that Dragon was originally designed mainly as a desktop application, with limited support for the web. I'm not sure what has changed in Dragon, I couldn't find any documentation related to the new behavior. It could also be that something changed at the operating system level in Windows. 

Regardless, if older versions of Dragon do not support standard keyboard events I'm not sure there's any reasonable workaround for that. I'd lean towards closing this issue. It can always be reopened if major barriers are reported again.
